src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -4493,7 +4493,7 @@ DataPartsVector MergeTreeData::grabActivePartsToRemoveForDropRange(
         /// It's a DROP PART and it's already executed by fetching some covering part
         bool is_drop_part = !drop_range.isFakeDropRangePart() && drop_range.min_block;
 
-        if (is_drop_part && (part->info.min_block != drop_range.min_block || part->info.max_block != drop_range.max_block || part->info.getMutationVersion() != drop_range.getMutationVersion()))
+        if (is_drop_part && (part->info.min_block != drop_range.min_block || part->info.max_block != drop_range.max_block))
         {
             /// Why we check only min and max blocks here without checking merge
             /// level? It's a tricky situation which can happen on a stale
```

Comment:
Maybe we need to update the comment, now we don't check level and mutation version